File: solutions/300.longest-increasing-subsequence/longest-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const lengthOfLIS = nums=> {
  if (!nums.length) return 0
  const record = [1, ...new Array(nums.length-1)]
  for (var i = 1; i < nums.length; i++) {
    record[i] = 1
    for (var j = 0; j < i; j++) {
      if (nums[i] > nums[j])
        record[i] = Math.max(record[i], record[j] + 1)
    }
  }
  return Math.max(...record)
}

File: solutions/300.longest-increasing-subsequence/longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        1 3 8 9 2 3

        1 2 3 4 1
        """
        if not nums:
            return 0
        
        dp = [1] * len(nums)

        for i in range(len(nums)):
            for j in range(i):
                if nums[i] > nums[j]:
                    dp[i] = max(dp[i], dp[j] + 1)

        return max(dp)

